packages/site/cms-site/src/iframebridge/IFrameBridge.tsx

```diff
@@ -84,24 +84,49 @@ export const IFrameBridgeProvider = ({ children }: PropsWithChildren) => {
     const [previewElementsData, setPreviewElementsData] = useState<OverlayElementData[]>([]);
 
     const childrenWrapperRef = useRef<HTMLDivElement>(null);
+    const childrenWrapperWidth = childrenWrapperRef.current?.offsetWidth;
 
     const recalculatePreviewElementsData = useCallback(() => {
+        if (!childrenWrapperWidth) {
+            return;
+        }
+
         const newPreviewElementsData = previewElements
             .map((previewElement) => {
                 const childNodes = getRecursiveChildrenOfPreviewElement(previewElement.element);
                 const positioning = getCombinedPositioningOfElements(childNodes);
 
+                const isRenderedOutsideOfViewportWidth = positioning.left > childrenWrapperWidth;
+                const spaceBetweenRightEdgeOfElementAndRightEdgeOfViewport = positioning.left + positioning.width - childrenWrapperWidth;
+                const tooWideForPreviewViewportByPixels =
+                    spaceBetweenRightEdgeOfElementAndRightEdgeOfViewport > 0 ? spaceBetweenRightEdgeOfElementAndRightEdgeOfViewport : 0;
+
+                if (isRenderedOutsideOfViewportWidth) {
+                    // TODO: Simply return `null` here after updating to typescript 5+
```

Comment:
Why is this related to TypeScript 5?

Reply:
Newer TS understands that when I filter by `!== null` that the value can not be `null` in the following `.sort()` and `.map()`. 

At least I accidentally had TS 5.x set in my IDE and I didn't see the error there and hovering over the value explicitly didn't show `null` as a possible type of the value.